### DISCUSSION TITLE
Accessing ax package after installing on Thonny shell

### DISCUSSION THREAD

**Discussion by aclasky** [AccelerationConsortium/ac-microcourses#106] [Hello World] [answered]
I am working on the module 1.3 Bayesian Optimization. I worked through the tutorial on Google Colab but wanted to make sure I could run it on my own machine too.
To install ax-platform I went into Thonny > Tools > Open system shell > pip install ax-platform. This installed successfully. Then, when I go to run the example code from the course, I get an ImportError.

![image](https://github.com/user-attachments/assets/4818512e-0f12-4df6-803c-3e209675f274)

I tried restarting Thonny, but the error is still present. I can see ax on my site-packages folder:
![image](https://github.com/user-attachments/assets/d4054e0a-9eba-434e-b2d1-0ec57847122c)

It is also present in the plug-ins list, but I think that is separate from the packages. Tools > Manage packages is greyed out unless I am using the Pico microcontroller.
![image](https://github.com/user-attachments/assets/b3df11f5-53c8-4aa3-8f8d-0e92bfa09d7f)

Any help would be much appreciated! I have a feeling Thonny just isn't "seeing" the installed package but I'm not sure how to direct it appropriately.

Thank you!

**Comment by sgbaird** [comment, marked as answer]
My suspicion is that the command is being sent to the microcontroller rather than to Python. (For context, Ax isn't something that can be installed onto the microcontroller directly). On the bottom-right of your Thonny window, check to see which environment is selected. If it's the MicroPython / microcontroller environment, try switching to the Python one.

As an aside, you might consider trying out VS Code, which is nearly identical to the GitHub Codespaces interface you'd be accustomed to from the course. Just requires a bit of extra setup.

**Reply by sgbaird** [reply]
There's plenty of documentation elsewhere, but this gets covered briefly in a later course: https://ac-microcourses.readthedocs.io/en/latest/courses/software-dev/4.2-vscode-setup.html.

**Reply by aclasky** [reply]
Thanks for your help! Yes I was operating in the wrong environment and didn't think to check. The code runs without error now.
